src/lib/problem_data.c

```diff
@@ -263,6 +263,35 @@ char *problem_data_get_content_or_NULL(problem_data_t *problem_data, const char
     return item->content;
 }
 
+char *problem_data_get_sub_components(const char *component)
+{
+    const char *array[COMPONENTS_WITH_SUBCOMPONENTS][2] = 
```

Comment:
You should be able to leave out the size of the first dimension unspecified. You can then use the `G_N_ELEMENTS` macro, compute it yourself or use some sort of null terminator.